fastFM/ffm2.pyx

```diff
@@ -77,11 +78,35 @@ def ffm_predict(np.ndarray[np.float64_t, ndim = 1] w_0,
 
     return y
 
+# The main piece of the glue between Python, Cython and C++
+# In essence it wraps the python function so it can be used in C++ space.
+# Convert the python function from a pointer back into a python object and invoke
+# with other parameters. (For now just one, `current_iter`)
+# TODO: Handle exceptions in a better manner
+cdef bool fit_callback_wrapper(string json_str, void* python_function):
```

Comment:
We could return string instead of bool, this way we can send more informative error messages back or other info like mcmc stats as you have mentioned.